build/charts/antrea/crds/bgppolicy.yaml

```diff
@@ -0,0 +1,129 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: bgppolicies.crd.antrea.io
+spec:
+  group: crd.antrea.io
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - nodeSelector
+                - advertisements
+              properties:
+                nodeSelector:
+                  type: object
+                  properties:
+                    matchExpressions:
+                      items:
+                        properties:
+                          key:
+                            type: string
+                          operator:
+                            enum:
+                              - In
+                              - NotIn
+                              - Exists
+                              - DoesNotExist
+                            type: string
+                          values:
+                            items:
+                              type: string
+                              pattern: "^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"
+                            type: array
+                        type: object
+                      type: array
+                    matchLabels:
+                      x-kubernetes-preserve-unknown-fields: true
+                localASN:
+                  type: integer
+                  format: int32
+                  minimum: 64512
+                  maximum: 65535
+                  default: 64512
+                listenPort:
+                  type: integer
+                  format: int32
+                  minimum: 1
+                  maximum: 65535
+                  default: 179
+                advertisements:
+                  type: object
+                  properties:
+                    service:
+                      type: object
+                      anyOf:
+                        - required: [ clusterIPs ]
+                        - required: [ externalIPs ]
+                        - required: [ loadBalancerIPs ]
```

Comment:
If you want to keep this requirement, you should add a comment in pkg/apis/crd/v1alpha1/types.go for the `ServiceAdvertisement` struct, indicating that at least one of these fields should be set.

Personally, I wonder if we should remove this requirement. It feels strange that `service: {}` is not accepted, while `service: { clusterIPs: false }` is valid.
I do understand where you are coming from, as `services: {}` would not trigger any IP advertisement, while `pods: {}` would advertise all Pod IPs, which seems like a discrepancy. Still, I think it may be better than the current requirement.

Reply:
I have removed the requirements and added some comments in `pkg/apis/crd/v1alpha1/types.go`. You could find it here: https://github.com/antrea-io/antrea/pull/6009/files#r1635761161